brevproduksjon/src/main/java/no/nav/foreldrepenger/fpformidling/brevproduksjon/mapper/innvilgelsefp/ForeldrepengerInnvilgelseDokumentdataMapper.java

```diff
@@ -210,31 +211,37 @@ private List<Utbetalingsperiode> finnInnvilgedePerioderMedUtbetaling(List<Utbeta
         return utbetalingsperioder.stream().filter(Utbetalingsperiode::isInnvilget).filter(up -> up.getPrioritertUtbetalingsgrad().erStørreEnnNull()).toList();
     }
 
-    private boolean vurderOmGraderingOgFulltUttak(List<Utbetalingsperiode> innvilgedeUtbetalingsperioder) {
-        if (innvilgedeUtbetalingsperioder != null) {
-            var arbeidsforhold = innvilgedeUtbetalingsperioder.stream()
-                    .flatMap(utbetalingsperiode -> utbetalingsperiode.getArbeidsforholdsliste().stream())
-                    .toList();
-            var annenAktivitet = innvilgedeUtbetalingsperioder.stream()
-                    .flatMap(utbetalingsperiode -> utbetalingsperiode.getAnnenAktivitetsliste().stream())
-                    .toList();
-            var næring = innvilgedeUtbetalingsperioder.stream().map(Utbetalingsperiode::getNæring).toList();
-
-            var gradering = arbeidsforhold.stream().filter(Objects::nonNull).anyMatch(Arbeidsforhold::isGradering)
-                    || annenAktivitet.stream().filter(Objects::nonNull).anyMatch(AnnenAktivitet::isGradering)
-                    || næring.stream().filter(Objects::nonNull).anyMatch(Næring::isGradering);
-
-            if (gradering) {
-                var fullUtbetalingArbforhold=  arbeidsforhold.stream().filter(Objects::nonNull).anyMatch(af -> af.getUtbetalingsgrad().equals(Prosent.HUNDRE));
-                var fullUtbetAnnenAktivitet = annenAktivitet.stream().filter(Objects::nonNull).anyMatch(aa -> aa.getUtbetalingsgrad().equals(Prosent.HUNDRE));
-                var fullUtbetalingNæring = næring.stream().filter(Objects::nonNull).anyMatch(n -> n.getUtbetalingsgrad().equals(Prosent.HUNDRE));
-
-                return fullUtbetalingArbforhold || fullUtbetAnnenAktivitet ||fullUtbetalingNæring;
-            }
+    private boolean vurderOmGraderingOgFulltUttakISammePeriode(Beregningsgrunnlag beregningsgrunnlag, int antallArbeidsgivere, List<Utbetalingsperiode> innvilgedeUtbetalingsperioder) {
+        if (innvilgedeUtbetalingsperioder != null && (harFlereAktivitetStatuser(beregningsgrunnlag) || antallArbeidsgivere > 1)) {
+            //Sjekk om gradering og fullt uttak i samme periode
+            return innvilgedeUtbetalingsperioder.stream()
```

Comment:
Tips Optional.ofNullable(innvilgedeUtbetalingsperioder).orElse(List.of()).stream()